sys/compat/cloudabi/cloudabi_sock.c

```diff
@@ -82,8 +84,11 @@ cloudabi_sock_recv(struct thread *td, cloudabi_fd_t fd, struct iovec *data,
 		.msg_iovlen = datalen,
 	};
 	struct mbuf *control;
+	cloudabi_fd_t * __capability fds;
 	int error;
 
+	fds = __USER_CAP(in_fds, fdslen * sizeof(int));
```

Comment:
Perhaps this should be:
```suggestion
	fds = __USER_CAP(in_fds, fdslen * sizeof(cloudabi_fd_t));
```
The two are the same, but this way we say what we really mean, and if something ever screws up and `cloudabi_fd_t` is anything but the `uint32_t` it currently is (or `int` ever is not 32-bit) then it'll trap with a length violation on one of the two buffers.

(or `sizeof(*fds)`)

Reply:
I had done this because the existing code uses sizeof(int) all over the place when calling copyin/copyout, etc.  It's not just here, but ubiquitous throughout the cloudabi layer when dealing with arrays of fds.

Reply:
Actually, it is really just this file, hmm.  Some of the sizeof(int) is because it is working with the native cmsg data.  I can perhaps use `__USE_CAP_ARRAY` here just fine.